oak_runtime/src/node/wasm/mod.rs

```diff
@@ -471,6 +474,165 @@ impl WasmInterface {
 
         Ok(())
     }
+
+    /// Corresponds to the host ABI function [`channel_label`](https://github.com/project-oak/oak/blob/main/docs/abi.md#channel_label).
+    fn channel_label(
+        &mut self,
+        handle: oak_abi::Handle,
+
+        dest: AbiPointer,
+        dest_capacity: AbiPointerOffset,
+        actual_length_addr: AbiPointer,
+    ) -> Result<(), OakStatus> {
+        trace!(
+            "{}: channel_label({}, {}, {}, {})",
+            self.pretty_name,
+            handle,
+            dest,
+            dest_capacity,
+            actual_length_addr
+        );
+
+        self.validate_ptr(dest, dest_capacity)?;
+
+        let label = self
+            .runtime
+            .get_serialized_channel_label(handle, dest_capacity as usize)?;
+
+        let actual_length = match &label {
+            LabelSerializationStatus::Success(label) => label.len(),
+            LabelSerializationStatus::NeedsCapacity(a) => *a,
+        };
+
+        let raw_writer = &mut [0; 4];
+        LittleEndian::write_u32(raw_writer, actual_length as u32);
+        self.get_memory()
+            .set(actual_length_addr, raw_writer)
+            .map_err(|err| {
+                error!(
+                    "{}: channel_label(): Unable to write actual length into guest memory: {:?}",
+                    self.pretty_name, err
+                );
+                OakStatus::ErrInvalidArgs
+            })?;
+
+        match label {
+            LabelSerializationStatus::Success(label) => {
+                self.get_memory().set(dest, &label).map_err(|err| {
+                    error!(
+                        "{}: channel_label(): Unable to write destination buffer into guest memory: {:?}",
+                        self.pretty_name, err
+                    );
+                    OakStatus::ErrInvalidArgs
+                })
+            }
+            LabelSerializationStatus::NeedsCapacity(_) => Err(OakStatus::ErrBufferTooSmall),
+        }
```

Comment:
Could we encapsulate this as `write_label_to_memory(&self, label: LabelSerializationStatus, dest: AbiPointer, dest_capacity: AbiPointerOffset, actual_length_addr: AbiPointer) -> Result<(), OakStatus>` and re-use in 3 places?